Tools integrating semgrep  programatically breaks as the API is being updated on releases.

**Is your feature request related to a problem? Please describe.**
njsscan was using semgrep 0.38 and a recent upgrade to 0.45 breaks tests

```
        output_handler = OutputHandler(
>           OutputSettings(
                output_format=OutputFormat.JSON,
                output_destination=None,
                error_on_findings=False,
                verbose_errors=False,
                strict=False,
                timeout_threshold=3,
                json_stats=False,
                output_per_finding_max_lines_limit=None,
            ),
            stdout=io_capture,
        )
E       TypeError: __new__() missing 1 required positional argument: 'json_time'

```

Current njsscan integration with semgrep: https://github.com/ajinabraham/libsast/blob/master/libsast/core_sgrep/helpers.py#L9

**Describe the solution you'd like**

Can we have two sets of APIs?

For seamless integration and maintenance, the semgrep API has to follow a contract with third-party tools. I suggest an internal and external API. The internal API changes and evolves with semgrep and the external one is fixed, documented, and changes only on major releases / a major security update.

**Describe alternatives you've considered**
NA

**Additional context**
NA

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -12,6 +12,10 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 
 ### Changed
 - .git/ directories are ignored when scanning
+- External Python API (`semgrep_main.invoke_semgrep`) now takes an
+  optional `OutputSettings` argument for controlling output
+- `OutputSettings.json_time` has moved to `OutputSettings.output_time`,
+  this and many other `OutputSettings` arguments have been made optional
 
 ## [0.49.0](https://github.com/returntocorp/semgrep/releases/tag/v0.49.0) - 2021-04-28
 

--- a/semgrep/semgrep/constants.py
+++ b/semgrep/semgrep/constants.py
@@ -38,7 +38,7 @@ class OutputFormat(Enum):
     VIM = auto()
 
     def is_json(self) -> bool:
-        return self == OutputFormat.JSON or self == OutputFormat.JSON_DEBUG
+        return self in [OutputFormat.JSON, OutputFormat.JSON_DEBUG, OutputFormat.SARIF]
 
 
 # Inline 'noqa' implementation modified from flake8:

--- a/semgrep/semgrep/output.py
+++ b/semgrep/semgrep/output.py
@@ -485,16 +485,19 @@ def build_vim_output(rule_matches: List[RuleMatch], rules: FrozenSet[Rule]) -> s
     return "\n".join(":".join(_get_parts(rm)) for rm in rule_matches)
 
 
+# WARNING: this class is unofficially part of our external API. It can be passed
+# as an argument to our official API: 'semgrep_main.invoke_semgrep'. Try to minimize
+# changes to this API, and make them backwards compatible, if possible.
 class OutputSettings(NamedTuple):
     output_format: OutputFormat
-    output_destination: Optional[str]
-    error_on_findings: bool
-    verbose_errors: bool
-    strict: bool
-    output_per_finding_max_lines_limit: Optional[int]
-    output_per_line_max_chars_limit: Optional[int]
-    json_stats: bool
-    output_time: bool
+    output_destination: Optional[str] = None
+    output_per_finding_max_lines_limit: Optional[int] = None
+    output_per_line_max_chars_limit: Optional[int] = None
+    error_on_findings: bool = False
+    verbose_errors: bool = False
+    strict: bool = False
+    json_stats: bool = False
+    output_time: bool = False
     timeout_threshold: int = 0
 
 
@@ -731,7 +734,7 @@ class OutputHandler:
         debug_steps = None
         if output_format == OutputFormat.JSON_DEBUG:
             debug_steps = self.debug_steps_by_rule
-        if output_format.is_json():
+        if output_format in [OutputFormat.JSON, OutputFormat.JSON_DEBUG]:
             return build_output_json(
                 self.rule_matches,
                 self.semgrep_structured_errors,

--- a/semgrep/semgrep/semgrep_main.py
+++ b/semgrep/semgrep/semgrep_main.py
@@ -5,8 +5,10 @@ from io import StringIO
 from pathlib import Path
 from re import sub
 from typing import Any
+from typing import Dict
 from typing import List
 from typing import Optional
+from typing import Union
 
 import attr
 
@@ -105,27 +107,22 @@ def rule_match_nosem(rule_match: RuleMatch, strict: bool) -> bool:
     return result
 
 
-def invoke_semgrep(config: Path, targets: List[Path], **kwargs: Any) -> Any:
+def invoke_semgrep(
+    config: Path,
+    targets: List[Path],
+    output_settings: Optional[OutputSettings] = None,
+    **kwargs: Any,
+) -> Union[Dict[str, Any], str]:
     """
-    Call semgrep with config on targets and return result as a json object
+    Return Semgrep results of 'config' on 'targets' as a dict|str
 
-    Uses default arguments of MAIN unless overwritten with a kwarg
+    Uses default arguments of 'semgrep_main.main' unless overwritten with 'kwargs'
     """
+    if output_settings is None:
+        output_settings = OutputSettings(output_format=OutputFormat.JSON)
+
     io_capture = StringIO()
-    output_handler = OutputHandler(
-        OutputSettings(
-            output_format=OutputFormat.JSON,
-            output_destination=None,
-            error_on_findings=False,
-            verbose_errors=False,
-            strict=False,
-            json_stats=False,
-            output_time=False,
-            output_per_finding_max_lines_limit=None,
-            output_per_line_max_chars_limit=None,
-        ),
-        stdout=io_capture,
-    )
+    output_handler = OutputHandler(output_settings, stdout=io_capture)
     main(
         output_handler=output_handler,
         target=[str(t) for t in targets],
@@ -135,7 +132,14 @@ def invoke_semgrep(config: Path, targets: List[Path], **kwargs: Any) -> Any:
         **kwargs,
     )
     output_handler.close()
-    return json.loads(io_capture.getvalue())
+
+    result: Union[Dict[str, Any], str] = (
+        json.loads(io_capture.getvalue())
+        if output_settings.output_format.is_json()
+        else io_capture.getvalue()
+    )
+
+    return result
 
 
 def main(

